tools/wake-format/emitter.cpp

```diff
@@ -337,16 +335,45 @@ wcl::doc Emitter::walk_token(ctx_t ctx, CSTElement node) {
 
 wcl::doc Emitter::walk_rhs(ctx_t ctx, CSTElement node) { return walk_placeholder(ctx, node); }
 
-wcl::doc Emitter::walk_apply(ctx_t ctx, CSTElement node) { return walk_placeholder(ctx, node); }
+wcl::doc Emitter::walk_apply(ctx_t ctx, CSTElement node) {
+  assert(node.id() == CST_APP);
+
+  return formatter()
+      .walk({CST_ID, CST_APP}, WALK(walk_node))
+      .consume_wsnl()
+      .space()
+      .walk(WALK(walk_node))
+      .format(ctx, node.firstChildElement());
+}
 
 wcl::doc Emitter::walk_arity(ctx_t ctx, CSTElement node) { return walk_placeholder(ctx, node); }
 
 wcl::doc Emitter::walk_ascribe(ctx_t ctx, CSTElement node) { return walk_placeholder(ctx, node); }
 
-wcl::doc Emitter::walk_binary(ctx_t ctx, CSTElement node) { return walk_placeholder(ctx, node); }
+wcl::doc Emitter::walk_binary(ctx_t ctx, CSTElement node) {
+  assert(node.id() == CST_BINARY);
+
+  return formatter()
+      .walk(WALK(walk_node))
+      .consume_wsnl()
+      .escape([this](wcl::doc_builder& bdr, ctx_t ctx, CSTElement& node) {
+        assert(node.id() == CST_OP);
+        // only add a space if the binop is not a comma
+        // a+b -> a + b
+        // a,b -> a, b
+        if (node.firstChildElement().id() != TOKEN_OP_COMMA) {
```

Comment:
In this PR don't special case anything I think. There's several special cases for operators. the "." operator for instance has no surrounding spaces.

Reply:
yeah, I removed it. I was just porting over the other code I wrote sitting in an abandoned branch so it could go away.